src/LineageTree/lineageTree.py

```diff
@@ -2400,6 +2403,595 @@ def get_cells_at_t_from_root(self, r: [int, list], t: int = None) -> list:
+            # while i > start_d or j > 0:
+            tb_type = traceback_mat[i, j]
+            if tb_type == 0:
+                # Match
+                i -= 1
+                j -= 1
+            elif tb_type == 1:
+                # Insertion
+                i -= 1
+            elif tb_type == 2:
+                # Deletion
+                j -= 1
+            path.append((i, j))
+
+        # Strip infinity edges from cost_mat before returning
+        cost_mat = cost_mat[1:, 1:]
+        return (path[::-1], cost_mat)
+
+    @staticmethod
+    def __rigid_transform_3D(A, B):
+        assert A.shape == B.shape
+
+        num_rows, num_cols = A.shape
+        if num_rows != 3:
+            raise Exception(
+                f"matrix A is not 3xN, it is {num_rows}x{num_cols}"
+            )
+
+        num_rows, num_cols = B.shape
+        if num_rows != 3:
+            raise Exception(
+                f"matrix B is not 3xN, it is {num_rows}x{num_cols}"
+            )
+
+        # find mean column wise
+        centroid_A = np.mean(A, axis=1)
+        centroid_B = np.mean(B, axis=1)
+
+        # ensure centroids are 3x1
+        centroid_A = centroid_A.reshape(-1, 1)
+        centroid_B = centroid_B.reshape(-1, 1)
+
+        # subtract mean
+        Am = A - centroid_A
+        Bm = B - centroid_B
+
+        H = Am @ np.transpose(Bm)
+
+        # find rotation
+        U, S, Vt = np.linalg.svd(H)
+        R = Vt.T @ U.T
+
+        # special reflection case
+        if np.linalg.det(R) < 0:
+            # print("det(R) < R, reflection detected!, correcting for it ...")
+            Vt[2, :] *= -1
+            R = Vt.T @ U.T
+
+        t = -R @ centroid_A + centroid_B
+
+        return R, t
+
+    def __interpolate(self, track1, track2, threshold):
+        """
+        Interpolate two series that have different lengths
+
+            Args:
+                track1 (list) and track2 (list): list of nodes in cell cycle
+                threshold (int): set a maximum number of points a track can have
+
+            Returns:
+                (list of list) x, y, z postions for track1
+                (list of list) x, y, z postions for track2
+        """
+        inter1_pos = []
+        inter2_pos = []
+
+        track1_pos = np.array([self.pos[c_id] for c_id in track1])
+        track2_pos = np.array([self.pos[c_id] for c_id in track2])
+
+        # Both tracks have the same length and size below the threshold - nothing is done
+        if len(track1) == len(track2) and (
+            len(track1) <= threshold or len(track2) <= threshold
+        ):
+            return track1_pos, track2_pos
+        # Both tracks have the same length but one or more sizes are above the threshold
+        elif (
+            len(track1) == len(track2)
+            and (len(track1) > threshold or len(track2) > threshold)
+            or len(track1) != len(track2)
+            and (len(track1) > threshold or len(track2) > threshold)
+        ):
```

Comment:
You are doing weird comparisons:
```python
            len(track1) == len(track2)
            and (len(track1) > threshold or len(track2) > threshold)
            or len(track1) != len(track2)
            and (len(track1) > threshold or len(track2) > threshold)
```
is equivalent to:
```python
            (len(track1) > threshold or len(track2) > threshold)
            and (
                        len(track1) != len(track2)
                        or len(track1) == len(track2)
            )
```
and because 
```python
            (
                        len(track1) != len(track2)
                        or len(track1) == len(track2)
            )
```
is always true then it is only:
```python
(len(track1) > threshold or len(track2) > threshold)
```
am I correct and if yes is it normal?